lib/vector-core/src/fanout.rs

```diff
@@ -473,20 +613,30 @@ mod tests {
 
             // Now our second send should actually be able to complete.  We'll assert that whichever
             // sender we removed does not get the next event:
-            let second_send_result = assert_ready!(second_send.poll());
-            assert!(second_send_result.is_ok());
+            assert_ready!(second_send.poll());
             drop(second_send);
 
-            let mut expected_next = [
-                Some(events[1].clone()),
-                Some(events[1].clone()),
-                Some(events[1].clone()),
-            ];
-            expected_next[sender_id] = None;
-
             for (i, receiver) in receivers.iter_mut().enumerate() {
-                assert_eq!(expected_next[i], receiver.next().await);
+                if i != sender_id {
+                    assert!(
+                        receiver.next().await.is_some(),
+                        "receiver {} is not sender_id {} and should have the event",
+                        i,
+                        sender_id
+                    );
+                }
             }
+
+            // let mut expected_next = [
+            // Some(events[1].clone()),
+            // Some(events[1].clone()),
+            // Some(events[1].clone()),
+            // ];
+            // expected_next[sender_id] = None;
+
+            // for (i, receiver) in receivers.iter_mut().enumerate() {
+            // assert_eq!(expected_next[i], receiver.next().await);
+            // }
```

Comment:
Intentionally left?

Reply:
Nope! I meant to go back and reevaluate what kind of assertions we can make here.

Reply:
Addressed in [710f261](https://github.com/vectordotdev/vector/pull/11348/commits/710f2610b6d0e1efa15a6ced09003602dc43e10a). This was a little trickier than I expected, but we can make basically the same assertions now.